WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApllicationLogDetailView.swift

```diff
@@ -0,0 +1,103 @@
+import SwiftUI
+
+/// Hosting controller that wraps an `ApplicationLogDetailView`
+///
+final class ApplicationLogDetailViewController: UIHostingController<ApplicationLogDetailView> {
+    init(with contents: String, for date: String) {
+        let viewModel = ApplicationLogViewModel(logText: contents)
+        super.init(rootView: ApplicationLogDetailView(viewModel: viewModel))
+        self.title = date
+        viewModel.present = { [weak self] vc in
+            self?.present(vc, animated: true, completion: nil)
+        }
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+struct ApplicationLogDetailView: View {
+    @ObservedObject var viewModel: ApplicationLogViewModel
+
+    private let dateFormatter: DateFormatter = {
```

Comment:
⚠️  Do you think it's worth passing this as a parameter or adding the `@State` property wrapper to it? I wonder how many times this gets re-created with the view? 🤔

Reply:
Moved this to the view model instead